.github/workflows/cdk-connector-compatibility-test.yml

```diff
@@ -0,0 +1,134 @@
+name: CDK Connector Compatibility Test
+
+on:
+  pull_request:
+    paths:
+      - 'airbyte-cdk/java/**/*'
+      - 'airbyte-cdk/bulk/**/*'
+  schedule:
+    - cron: '0 12 * * *'
+
+jobs:
+  generate-matrix:
+    name: Generate Connector Matrix
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout Airbyte
+        uses: actions/checkout@v4
+        with:
+          submodules: true # Needed for airbyte-enterprise connectors (no-op otherwise)
+          fetch-depth: 0
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: "zulu"
+          java-version: "21"
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          gradle-version: "8.14"
+
+      - name: Trigger changes on all connectors using bulk cdk # this is just an easy way to get a list of all bulk cdk connectors
+        run: ./gradlew upgradeCdk --cdkVersion=local
+
+      - name: Generate Connector Matrix from Changes
+        id: generate-matrix
+        run: echo "connectors_matrix=$(./poe-tasks/get-modified-connectors.sh --json --local-cdk)" | tee -a $GITHUB_OUTPUT
+
+    outputs:
+      connectors-matrix: ${{ steps.generate-matrix.outputs.connectors_matrix }}
+
+  connectors-test:
```

Comment:
I'm planning to open another PR to refactor since there's a ton of duplicated code across different workflows. e.g. the [ci checks workflow](https://github.com/airbytehq/airbyte/blob/5bec6d364d1ebd5a99942109803e10589b2f722a/.github/workflows/connector-ci-checks.yml#L1) is doing a lot of what we are doing here.